gel/_internal/_qbmodel/_abstract/_base.py

```diff
@@ -169,7 +169,7 @@ def __new__(
             # The class registry only tracks the canonical base instances,
             # which are the ones that directly declare 'tname__'
             (tname := getattr(reflection, "name", None)) is not None
-            and 'tname__' in namespace
+            and any('tname__' in base.__dict__ for base in bases)
```

Comment:
Sigh I guess this is fine.

Also update the comment above this, which is now wrong.

Add that `tname__` isn't in `namespace` itself, then, I think?
Since the base shape also has base shape children, and we don't want those to get picked up.
(I don't know if it would be a bug for them to get picked up here, since they'd be correctly overwritten by their defined-later real children, but still.)